src/test/util/setup_common.h

```diff
@@ -121,6 +121,11 @@ struct TestChain100Setup : public RegTestingSetup {
     CBlock CreateAndProcessBlock(const std::vector<CMutableTransaction>& txns,
                                  const CScript& scriptPubKey);
 
+    CMutableTransaction CreateValidMempoolTransaction(CTransactionRef input_transaction,
+                                                      int input_vout,
+                                                      CScript output_destination,
+                                                      CAmount output_amount = CAmount(1 * COIN));
```

Comment:
The implementation assumes that the key to spend `input_transaction:input_vout` is `this->coinbaseKey`. Indeed all callers of this in #21061 pass some tx from `this->m_coinbase_txns[]`.

This is not clear from the above prototype. Maybe document this explicitly or even better - pass the spend key here, together with `input_transaction` and `input_vout`.

Reply:
good point! extracted `input_signing_key` as a param to be passed in.